explainer.md

```diff
@@ -0,0 +1,195 @@
+# Compression Streams Explained
+27 August 2019
+
+
+## What’s all this then?
+
+The "gzip" and "deflate" compression algorithms are extensively used in the
+web platform, but up until now have not been exposed to JavaScript. Since
+compression is naturally a streaming process, it is a good match for the
+[WHATWG Streams](https://streams.spec.whatwg.org/) API.
+
+CompressionStream is used to compress a stream. It accepts ArrayBuffer or
+ArrayBufferView chunks, and outputs Uint8Array.
+
+DecompressionStream is used to decompress a stream. It accepts
+ArrayBuffer or ArrayBufferView chunks, and outputs Uint8Array.
+
+Both APIs satisfy the concept of a [transform
+stream](https://streams.spec.whatwg.org/#ts-model) from the WHATWG
+Streams Standard.
+
+
+## Goal
+
+The goal is to provide a JavaScript API for compressing and decompressing data
+in the "gzip" ([RFC1952](https://tools.ietf.org/html/rfc1952)) or "deflate"
+([RFC1950](https://www.ietf.org/rfc/rfc1950.txt)) formats.
+
+
+## Non-goals
+
+*   Compression formats other than "gzip" and "deflate" will not be
+    supported in the first version of the API.
+*   Support for synchronous compression.
+
+
+## Motivation
+
+Existing libraries can be used to perform compression in the browser, however
+there are a number of benefits to a built-in capability.
+
+*   A built-in API does not need to be downloaded. This is particularly relevant
+    when compression is being used with the goal of reducing bandwidth usage.
+*   Native compression libraries are faster and use less power than JavaScript
+    or WASM libraries.
+*   Ergonomics is improved by having a standard, web-like API. By using streams
+    composability with other web platforms is improved.
+*   The web platform is unusual in not having native support for compression.
```

Comment:
As above, citation needed. Could point to other comparable platforms that ship with compression libraries by default.